ambry-router/src/main/java/com.github.ambry.router/NonBlockingRouterMetrics.java

```diff
@@ -31,42 +37,139 @@
   // @todo: Ensure all metrics here get updated appropriately.
   // @todo: chunk filling rate metrics.
   // @todo: More metrics for the RequestResponse handling (poll, handleResponse etc.)
-  // Rates
+
+  // Operation rate.
   public final Meter putBlobOperationRate;
+  public final Meter getBlobInfoOperationRate;
+  public final Meter getBlobOperationRate;
+  public final Meter deleteBlobOperationRate;
   public final Meter operationQueuingRate;
   public final Meter operationDequeuingRate;
 
+  // Latency.
   public final Histogram putBlobOperationLatencyMs;
   public final Histogram putChunkOperationLatencyMs;
+  public final Histogram getBlobInfoOperationLatencyMs;
+  public final Histogram getBlobOperationLatencyMs;
+  public final Histogram deleteBlobOperationLatencyMs;
   public final Histogram routerRequestLatencyMs;
 
-  // Operation Errors
-  public final Counter slippedPutSuccessCount;
+  // Operation error count.
   public final Counter putBlobErrorCount;
+  public final Counter getBlobInfoErrorCount;
+  public final Counter getBlobErrorCount;
+  public final Counter deleteBlobErrorCount;
   public final Counter operationAbortCount;
 
-  // Gauge
+  // Count for various errors.
+  public final Counter ambryUnavailableErrorCount;
+  public final Counter invalidBlobIdErrorCount;
+  public final Counter invalidPutArgumentErrorCount;
+  public final Counter operationTimedOutErrorCount;
+  public final Counter routerClosedErrorCount;
+  public final Counter unexpectedInternalErrorCount;
+  public final Counter blobTooLargeErrorCount;
+  public final Counter badInputChannelErrorCount;
+  public final Counter insufficientCapacityErrorCount;
+  public final Counter blobDeletedErrorCount;
+  public final Counter blobDoesNotExistErrorCount;
+  public final Counter blobExpiredErrorCount;
+  public final Counter unknownReplicaResponseError;
+  public final Counter closeErrorCount;
+  public final Counter unknownErrorCountForOperation;
+
+  // Misc metrics.
+  public final Meter operationErrorRate;
+  public final Counter slippedPutSuccessCount;
```

Comment:
What does this metric mean? How many times we have tried slipped puts and succeeded?